AppAuth.podspec

```diff
@@ -33,16 +33,25 @@ It follows the OAuth 2.0 for Native Apps best current practice
   s.platforms    = { :ios => "7.0", :osx => "10.9", :watchos => "2.0", :tvos => "9.0" }
 
   s.source       = { :git => "https://github.com/openid/AppAuth-iOS.git", :tag => s.version }
-
-  s.source_files = "Source/*.{h,m}"
   s.requires_arc = true
 
-  # iOS
-  s.ios.source_files      = "Source/iOS/**/*.{h,m}"
-  s.ios.deployment_target = "7.0"
-  s.ios.frameworks        = "SafariServices", "AuthenticationServices"
-
-  # macOS
-  s.osx.source_files = "Source/macOS/**/*.{h,m}"
-  s.osx.deployment_target = '10.9'
+  # Subspec for the core AppAuth library classes only, suitable for extensions.
+  s.subspec 'Core' do |core|
+     core.source_files = "Source/*.{h,m}"
+  end
+
+  # Subspec for the full AppAuth library, including platform-dependant external user agents.
+  s.subspec 'ExternalUserAgent' do |externalUserAgent|
+    externalUserAgent.dependency 'AppAuth/Core'
+    externalUserAgent.xcconfig = { 'GCC_PREPROCESSOR_DEFINITIONS' => %w($(inherited) APPAUTH_EXTERNALUSERAGENT=1) }
```

Comment:
seems like cocoapods doesn't like this line if the pod is fetched from github:
- `pod install` in the `Example-iOS_ObjC` example works just fine
- but it fails if I change 
  ```
  pod 'AppAuth', :path => '../../'
  ```
  to
  ```
  pod 'AppAuth', :git => 'https://github.com/WilliamDenniss/AppAuth-iOS.git', :branch => 'extensions'
  ```

It works if I change branch to `master`. The error message I see during `pod install` is
```
NoMethodError - undefined method `strip' for ["$(inherited)", "APPAUTH_EXTERNALUSERAGENT=1"]:Array
/Library/Ruby/Gems/2.3.0/gems/xcodeproj-1.6.0/lib/xcodeproj/config.rb:299:in `block in merge_attributes!'
/Library/Ruby/Gems/2.3.0/gems/xcodeproj-1.6.0/lib/xcodeproj/config.rb:297:in `merge!'
/Library/Ruby/Gems/2.3.0/gems/xcodeproj-1.6.0/lib/xcodeproj/config.rb:297:in `merge_attributes!'
/Library/Ruby/Gems/2.3.0/gems/xcodeproj-1.6.0/lib/xcodeproj/config.rb:212:in `merge!'
/Library/Ruby/Gems/2.3.0/gems/cocoapods-1.5.3/lib/cocoapods/generator/xcconfig/pod_xcconfig.rb:67:in `block in generate'
/Library/Ruby/Gems/2.3.0/gems/cocoapods-1.5.3/lib/cocoapods/generator/xcconfig/pod_xcconfig.rb:66:in `each'
/Library/Ruby/Gems/2.3.0/gems/cocoapods-1.5.3/lib/cocoapods/generator/xcconfig/pod_xcconfig.rb:66:in `generate'
/Library/Ruby/Gems/2.3.0/gems/cocoapods-1.5.3/lib/cocoapods/generator/xcconfig/pod_xcconfig.rb:39:in `save_as'
/Library/Ruby/Gems/2.3.0/gems/cocoapods-1.5.3/lib/cocoapods/installer/xcode/pods_project_generator/target_installer.rb:107:in `update_changed_file'
/Library/Ruby/Gems/2.3.0/gems/cocoapods-1.5.3/lib/cocoapods/installer/xcode/pods_project_generator/pod_target_installer.rb:384:in `create_xcconfig_file'
/Library/Ruby/Gems/2.3.0/gems/cocoapods-1.5.3/lib/cocoapods/installer/xcode/pods_project_generator/pod_target_installer.rb:28:in `block in install!'
/Library/Ruby/Gems/2.3.0/gems/cocoapods-1.5.3/lib/cocoapods/user_interface.rb:142:in `message'
/Library/Ruby/Gems/2.3.0/gems/cocoapods-1.5.3/lib/cocoapods/installer/xcode/pods_project_generator/pod_target_installer.rb:19:in `install!'
/Library/Ruby/Gems/2.3.0/gems/cocoapods-1.5.3/lib/cocoapods/installer/xcode/pods_project_generator.rb:176:in `block (2 levels) in install_libraries'
/Library/Ruby/Gems/2.3.0/gems/cocoapods-1.5.3/lib/cocoapods/installer/xcode/pods_project_generator.rb:173:in `each'
/Library/Ruby/Gems/2.3.0/gems/cocoapods-1.5.3/lib/cocoapods/installer/xcode/pods_project_generator.rb:173:in `block in install_libraries'
/Library/Ruby/Gems/2.3.0/gems/cocoapods-1.5.3/lib/cocoapods/user_interface.rb:142:in `message'
/Library/Ruby/Gems/2.3.0/gems/cocoapods-1.5.3/lib/cocoapods/installer/xcode/pods_project_generator.rb:167:in `install_libraries'
/Library/Ruby/Gems/2.3.0/gems/cocoapods-1.5.3/lib/cocoapods/installer/xcode/pods_project_generator.rb:65:in `generate!'
/Library/Ruby/Gems/2.3.0/gems/cocoapods-1.5.3/lib/cocoapods/installer.rb:184:in `block in generate_pods_project'
/Library/Ruby/Gems/2.3.0/gems/cocoapods-1.5.3/lib/cocoapods/user_interface.rb:64:in `section'
/Library/Ruby/Gems/2.3.0/gems/cocoapods-1.5.3/lib/cocoapods/installer.rb:183:in `generate_pods_project'
/Library/Ruby/Gems/2.3.0/gems/cocoapods-1.5.3/lib/cocoapods/installer.rb:119:in `install!'
/Library/Ruby/Gems/2.3.0/gems/cocoapods-1.5.3/lib/cocoapods/command/install.rb:41:in `run'
/Library/Ruby/Gems/2.3.0/gems/claide-1.0.2/lib/claide/command.rb:334:in `run'
/Library/Ruby/Gems/2.3.0/gems/cocoapods-1.5.3/lib/cocoapods/command.rb:52:in `run'
/Library/Ruby/Gems/2.3.0/gems/cocoapods-1.5.3/bin/pod:55:in `<top (required)>'
/usr/local/bin/pod:22:in `load'
/usr/local/bin/pod:22:in `<main>'
```

Reply:
oh yes, you're right @julienbodet - i used the wrong local AppAuth copy when I tested it this is why it worked...

Reply:
changing it to `    externalUserAgent.xcconfig = { 'GCC_PREPROCESSOR_DEFINITIONS' => '$(inherited) APPAUTH_EXTERNALUSERAGENT=1' }` seems to resolve that particular issue.